cloud/EAP7-1769-Helm-cahrts-Configuring-Kubernetes-startup-probe.adoc

```diff
@@ -0,0 +1,137 @@
+---
+categories:
+ - helm-charts
+stability-level: default
+issue: https://github.com/wildfly/wildfly-charts/issues/37
+feature-team:
+ developer: yersan
+ sme: jmesnil
+  -
+ outside-perspective:
+  - laDok8
+---
+= [Default] Add support for configuring Kubernetes startup probe via Helm Charts
+:author:            Yeray Borges
+:email:             yborgess@redhat.com
+:toc:               left
+:icons:             font
+:idprefix:
+:idseparator:       -
+
+== Overview
+
+The probes (Liveness and Readiness) added by the Helm Chart uses the following default values:
+
+[literal]
+failureThreshold: 3
+periodSeconds: 10
+successThreshold: 1
+timeoutSeconds: 1
+initialDelaySeconds: 0
+
+For the liveness probes, for example, this configuration means the server could take until `failureThreshold*periodSeconds=30` seconds to get started. If such a condition doesn't occur the cluster will kill the pod to start over it again (restart policy is always).
+
+Sometimes the pod could need more than 30 seconds to start, for example:
+
+The user has registered other liveness checks in their application code and to satisfy those conditions is needed more time.
+The management operations executed at runtime when the pod is started could also take some time to be applied, which delays the availability of the endpoint used by the default liveness probe.
+The node where the pod is running is slow.
+In such a case, users could be tempted to increase the `initialDelaySeconds` or the `failureThreshold` configuration, however increasing the `initialDelaySeconds` will delay a fixed value, which could compromise the expected fast response. Increasing the `failureThreshold` could also break the established conditions you have decided to check your application at runtime.
+
+The alternative to those issues is to have the ability to configure a Startup Probe. The Startup Probe can be used to delay the initial checks done by the liveness and readiness probes, allowing some configuration without compromising them.
+
+The goal of this RFE is to define the default initial configuration of a Startup Probe so users can tweak it when required.
+
```

Comment:
```suggestion
For consistency sake same configuration shall also be introduced for liveness and readiness probes
```

Reply:
@laDok8 Notice that there was already a default configuration for the liveness and readiness probes in the Helm Chart. See the changes we added in this proposal: https://github.com/wildfly/wildfly-charts/pull/34/files

I don't mind tracking them as they were added with this proposal, and indeed, it is completely harmless and only paperwork, but I want to be clear that there were already default values for the aforementioned probes.

Are you following me on this? If so, to avoid any bikeshedding on this, I plan to accept your good suggestions, but first I would like to be sure we are on the same page about this.

Reply:
Ok, no objections from my side as long as we are on the same page.